server-worker/src/main/java/com/aliyun/emr/rss/service/deploy/worker/storage/PartitionFilesSorter.java

```diff
@@ -627,5 +635,11 @@ private void deleteOriginFiles() throws IOException {
         logger.warn("clean origin file failed, origin file is : {}", originFilePath);
       }
     }
+
+    private ByteBuffer expandBufferAndUpdateMemoryTracker(int oldCapacity, int newCapacity) {
+      memoryTracker.releaseSortMemory(oldCapacity);
+      memoryTracker.reserveSortMemory(newCapacity);
+      return ByteBuffer.allocateDirect(newCapacity);
```

Comment:
I have added cleaner().clean() before expandBufferAndUpdateMemoryTracker(), i think we still need to update memoryTracker otherwise its value won't be consistent.